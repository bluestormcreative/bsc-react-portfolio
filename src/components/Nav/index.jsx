import React from 'react';

const Nav = () => (
	<nav className="site-nav border-b-2 border-solid border-gray-400">
		About
		Portfolio
		Resume
		Contact
	</nav>
);

export default Nav;
