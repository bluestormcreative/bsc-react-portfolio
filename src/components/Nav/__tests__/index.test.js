import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

const selectedNavItem = 'About';
const mockSetSelectedNavItem = jest.fn();

afterEach(cleanup);

describe('Nav component', () => {
  it('renders', () => {
    render(
      <Nav
        selectedNavItem={selectedNavItem}
        setSelectedNavItem={mockSetSelectedNavItem}
      />
    );
  });

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(
      <Nav
        selectedNavItem={selectedNavItem}
        setSelectedNavItem={mockSetSelectedNavItem}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('links are visible', () => {
  it('inserts text into the links', () => {
    const { getByTestId } = render(
      <Nav
        selectedNavItem={selectedNavItem}
        setSelectedNavItem={mockSetSelectedNavItem}
      />
    );
    expect(getByTestId('portfolio')).toHaveTextContent('Portfolio');
    expect(getByTestId('about')).toHaveTextContent('About');
  });
});

describe('Click event sets active link', () => {
  it('sets classname nav-active on the clicked link', () => {
    const { getByTestId } = render(
      <Nav
        selectedNavItem={selectedNavItem}
        setSelectedNavItem={mockSetSelectedNavItem}
      />
    );
    fireEvent.click(getByTestId('about'));
    expect(getByTestId('about').classList.contains('nav-active')).toBe(true);
  });
});

describe('Click Event', () => {
  it('calls onClick handler and sets state', () => {
    const { getByTestId } = render(
      <Nav
        selectedNavItem={selectedNavItem}
        setSelectedNavItem={mockSetSelectedNavItem}
      />
    );
    fireEvent.click(getByTestId('portfolio'));
    expect(mockSetSelectedNavItem).toHaveBeenCalledTimes(1);
  });
})  
