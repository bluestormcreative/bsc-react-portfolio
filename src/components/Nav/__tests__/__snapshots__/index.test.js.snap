// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <nav
    class="site-nav border-b-2 border-solid border-gray-400 py-4 flex space-x-4"
  >
    <a
      class="nav-item inline-block nav-active"
      data-testid="about"
      href="#about"
    >
      About
    </a>
    <span
      class="nav-item inline-block "
      data-testid="portfolio"
    >
      Portfolio
    </span>
    <span
      class="nav-item inline-block "
    >
      Resume
    </span>
    <span
      class="nav-item inline-block "
    >
      Contact
    </span>
  </nav>
</DocumentFragment>
`;
